libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -44,3 +44,54 @@ int32_t AP_BattMonitor_Backend::get_capacity() const
 {
     return _mon.pack_capacity_mah(_state.instance);
 }
+
+// update battery resistance estimate
+// faster rates of change of the current and voltage readings cause faster updates to the resistance estimate
+// the battery resistance is calculated by comparing the latest current and voltage readings to a low-pass filtered current and voltage
+// high current steps are integrated into the resistance estimate by varying the time constant of the resistance filter
+void AP_BattMonitor_Backend::update_resistance_estimate()
+{
+    // return immediately if no current
+    if (!has_current() || _state.current_amps <= 0.0f) {
+        return;
+    }
+
+    // update maximum current seen since startup and protect against divide by zero
+    _current_max_amps = MAX(_current_max_amps, _state.current_amps);
+    float current_delta = _state.current_amps - _current_filt_amps;
+    if (is_zero(_current_max_amps) || is_zero(current_delta)) {
```

Comment:
If you change the initial entry function slightly as suggested then ```_current_max_amps``` must be > 0 by the time you get here, and you can discard the check for it.